### PR TITLE
fix(autoware_motion_velocity_planner_node): fix unreadVariable

### PR DESCRIPTION
**Pull request by kobayu858**
## Description
This is a fix based on cppcheck unreadVariable warnings.

```
planning/motion_velocity_planner/autoware_motion_velocity_planner_node/src/node.cpp:383:29: style: Variable 'input_trajectory_points' is assigned a value that is never used. [unreadVariable]
    input_trajectory_points = smooth_trajectory(input_trajectory_points, planner_data_);
                            ^
```

## Related links

**Parent Issue:**

- Link

## Notes for reviewers

None.

## Interface changes

None.

## Effects on system behavior

None.

### CODE REVIEW

**Comment by kobayu858** [comment]
Since `input_trajectory_points` was not used afterwards, the entire `if` statement was removed.

If it is left intentionally, it can be suppressed with a comment.

**Review by maxime-clem — APPROVED** [review]
This issue will be fixed in ~https://github.com/autowarefoundation/autoware.universe/pull/8384~
EDIT: new PR is here https://github.com/autowarefoundation/autoware.universe/pull/8453
But it if fine to merge your PR first if you want.

File: planning/motion_velocity_planner/autoware_motion_velocity_planner_node/src/node.cpp [hunk 1]
```diff
@@ -379,8 +379,6 @@ autoware_planning_msgs::msg::Trajectory MotionVelocityPlannerNode::generate_traj
 {
   autoware_planning_msgs::msg::Trajectory output_trajectory_msg;
   output_trajectory_msg.points = {input_trajectory_points.begin(), input_trajectory_points.end()};
-  if (smooth_velocity_before_planning_)
```
Comment (takayuki5168) [inline planning/motion_velocity_planner/autoware_motion_velocity_planner_node/src/node.cpp:382]: @maxime-clem This member variable can be removed?

Comment (maxime-clem) [reply planning/motion_velocity_planner/autoware_motion_velocity_planner_node/src/node.cpp:382]: It is correctly used in https://github.com/autowarefoundation/autoware.universe/pull/8453
Unless you mean that we should always smooth the velocity before planning, and so we can remove this parameter completely ?

**Comment by veqcc** [comment]
@maxime-clem 
May I close this PR?
I confirmed your PR https://github.com/autowarefoundation/autoware.universe/pull/8453 has fixed the `unreadVariable` warning!!

**Comment by maxime-clem** [comment]
> @maxime-clem May I close this PR? I confirmed your PR #8453 has fixed the `unreadVariable` warning!!

Thank you. I am closing the PR since the issue was already fixed.

**Pull request closed without merging**